### DISCUSSION TITLE
How to tell CIBW to build once, but test wheels on all future minor Python versions?

### DISCUSSION THREAD

**Discussion by astro-stan** [pypa/cibuildwheel#1903] [Q&A] [answered]
### Description

I want to build ABI3 (limited API) wheels with CIBW and I would like to build the wheel only on the lowest Python version I support but tell CIBW to test it on all future Python versions for a given platform.

Concretely, I am targeting 3.7+. I want to build wheels for Python 3.7 only (for all major platforms) but test them on 3.7-3.12 (for each platform).

I've been thinking I can maybe use `nox` for this, but that still requires all of the desired Python interpreters be installed. 

Is there a way to achieve this with CIBW?

**Comment by henryiii** [comment, marked as answer]
Yes, it does this out of the box. If it builds an ABI3 wheel, each subsequent run with a higher Python version will simply use that wheel and test it.

**Comment by henryiii** [comment]
(As long as you do have a test-command set up, of course)

**Comment by astro-stan** [comment]
Ah yes, indeed it does!

Sorry, this wasn't clear (to me) from the documentation.

Perhaps it's worth stating it explicitly somewhere (unless its already stated and I just didn't see it)?
